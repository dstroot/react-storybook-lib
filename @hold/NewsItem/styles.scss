a.news-link {
  text-decoration: none;
  color: inherit;
  &:hover {
    color: #008ed7;
    border-color: #008ed7;
  }
}

.media-body {
  padding-left: 10px;
  height: 115px;
  border-left: 5px solid white;
  transition: 0.3s;
  &:hover {
    border-color: #008ed7;
  }
}

.display-5 {
  font-size: 3em;
  font-weight: 300;
  margin-bottom: -5px;
}

.display-6 {
  font-size: 2.5em;
  font-weight: 300;
}
