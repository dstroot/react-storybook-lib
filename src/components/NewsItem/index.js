import React from 'react';
import placeholder from './media/logo.png';

const textLink = {
  fontSize: `calc(14px + 1vw + 0.25vmin)`,
  lineHeight: '1.1',
  marginBottom: '0',
  fontWeight: '300',
};

// https://getbootstrap.com/docs/4.1/layout/media-object/
const NewsItem = ({ item }) => (
  <>
    <a className="news-link" href={item.url}>
      <div className="media">
        <img
          className="mr-3 d-none d-sm-block"
          src={item.image || placeholder}
          alt="Generic placeholder"
          width="175px"
          height="115px"
        />
        <div className="media-body">
          <p className="mt-0 text-muted">
            {item.date} / <strong>{item.category}</strong>
          </p>
          <p style={textLink}>{item.headline}</p>
        </div>
      </div>
    </a>
    <hr />
  </>
);

export default NewsItem;

// <h3 className="font-weight-light">{item.headline}</h3>
