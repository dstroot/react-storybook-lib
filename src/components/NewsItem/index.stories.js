import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import NewsItem from '../NewsItem';

const data = {
  id: '1',
  date: '11-14-18',
  image: '',
  headline:
    'Pacific Life Foundation Donates $250,000 to the American Red Cross for Victims of the California Wildfires',
  url:
    'https://www.pacificlife.com/press-releases/pacific-life-foundation-donates--250-000-to-the-american-red-cro.html',
  category: 'Company News',
};

// NewsItem section
storiesOf('NewsItem', module).add('show NewsItem', () => (
  <NewsItem item={data} />
));
