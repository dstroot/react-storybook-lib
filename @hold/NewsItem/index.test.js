import React from 'react';
import renderer from 'react-test-renderer';

import NewsItem from '../NewsItem';
const data = {
  id: '1',
  date: '11-14-18',
  image: '',
  headline:
    'Pacific Life Foundation Donates $250,000 to the American Red Cross for Victims of the California Wildfires',
  url:
    'https://www.pacificlife.com/press-releases/pacific-life-foundation-donates--250-000-to-the-american-red-cro.html',
  category: 'Company News',
};

describe('NewsItem', () => {
  it('it should render', () => {
    const component = renderer.create(<NewsItem item={data} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
