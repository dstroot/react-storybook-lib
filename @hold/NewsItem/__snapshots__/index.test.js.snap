// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItem it should render 1`] = `
Array [
  <a
    className="news-link"
    href="https://www.pacificlife.com/press-releases/pacific-life-foundation-donates--250-000-to-the-american-red-cro.html"
  >
    <div
      className="media"
    >
      <img
        alt="Generic placeholder"
        className="mr-3 d-none d-sm-block"
        height="115px"
        src="logo.png"
        width="175px"
      />
      <div
        className="media-body"
      >
        <p
          className="mt-0 text-muted"
        >
          11-14-18
           / 
          <strong>
            Company News
          </strong>
        </p>
        <p
          style={
            Object {
              "fontSize": "calc(14px + 1vw + 0.25vmin)",
              "fontWeight": "300",
              "lineHeight": "1.1",
              "marginBottom": "0",
            }
          }
        >
          Pacific Life Foundation Donates $250,000 to the American Red Cross for Victims of the California Wildfires
        </p>
      </div>
    </div>
  </a>,
  <hr />,
]
`;
